// ionic-angular/components/img/img.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'ionic-angular/components/img/img';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
import * as import5 from '@angular/core/src/linker/element';
import * as import7 from '@angular/core/src/linker/view_type';
import * as import8 from '@angular/core/src/linker/element_ref';
import * as import9 from 'ionic-angular/platform/platform';
import * as import10 from '@angular/core/src/zone/ng_zone';
import * as import11 from '@angular/core/src/security';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from '@angular/core/src/linker/component_factory';
export var Wrapper_Img = (function () {
    function Wrapper_Img(p0, p1, p2) {
        this.changed = false;
        this._src = import1.UNINITIALIZED;
        this._width = import1.UNINITIALIZED;
        this._height = import1.UNINITIALIZED;
        this._alt = import1.UNINITIALIZED;
        this._title = import1.UNINITIALIZED;
        this.context = new import0.Img(p0, p1, p2);
    }
    Wrapper_Img.prototype.check_src = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._src, currValue))) {
            this.changed = true;
            this.context.src = currValue;
            this._src = currValue;
        }
    };
    Wrapper_Img.prototype.check_width = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._width, currValue))) {
            this.changed = true;
            this.context.width = currValue;
            this._width = currValue;
        }
    };
    Wrapper_Img.prototype.check_height = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._height, currValue))) {
            this.changed = true;
            this.context.height = currValue;
            this._height = currValue;
        }
    };
    Wrapper_Img.prototype.check_alt = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._alt, currValue))) {
            this.changed = true;
            this.context.alt = currValue;
            this._alt = currValue;
        }
    };
    Wrapper_Img.prototype.check_title = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._title, currValue))) {
            this.changed = true;
            this.context.title = currValue;
            this._title = currValue;
        }
    };
    Wrapper_Img.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        if (!throwOnChange) {
            if ((view.numberOfChecks === 0)) {
                this.context.ngOnInit();
            }
        }
        return changed;
    };
    return Wrapper_Img;
}());
var renderType_Img_Host = null;
var _View_Img_Host0 = (function (_super) {
    __extends(_View_Img_Host0, _super);
    function _View_Img_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Img_Host0, renderType_Img_Host, import7.ViewType.HOST, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckAlways);
    }
    _View_Img_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-img', rootSelector, null);
        this._appEl_0 = new import5.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Img0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Img_0_4 = new Wrapper_Img(new import8.ElementRef(this._el_0), this.parentInjector.get(import9.Platform), this.parentInjector.get(import10.NgZone));
        this._appEl_0.initComponent(this._Img_0_4.context, [], compView_0);
        compView_0.create(this._Img_0_4.context, this.projectableNodes, null);
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Img_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.Img) && (0 === requestNodeIndex))) {
            return this._Img_0_4.context;
        }
        return notFoundResult;
    };
    _View_Img_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._Img_0_4.detectChangesInternal(this, this._el_0, throwOnChange)) {
            this._appEl_0.componentView.markAsCheckOnce();
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = this._Img_0_4.context._width;
        if (import2.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementStyle(this._el_0, 'width', ((this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_0) == null) ? null : this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_0).toString()));
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this._Img_0_4.context._height;
        if (import2.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementStyle(this._el_0, 'height', ((this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_1) == null) ? null : this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_1).toString()));
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Img_Host0;
}(import3.AppView));
function viewFactory_Img_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Img_Host === null)) {
        (renderType_Img_Host = viewUtils.createRenderComponentType('', 0, import12.ViewEncapsulation.None, [], {}));
    }
    return new _View_Img_Host0(viewUtils, parentInjector, declarationEl);
}
export var ImgNgFactory = new import13.ComponentFactory('ion-img', viewFactory_Img_Host0, import0.Img);
var styles_Img = [];
var renderType_Img = null;
var _View_Img0 = (function (_super) {
    __extends(_View_Img0, _super);
    function _View_Img0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Img0, renderType_Img, import7.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckOnce);
    }
    _View_Img0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'img-placeholder');
        this._expr_0 = import1.UNINITIALIZED;
        this._expr_1 = import1.UNINITIALIZED;
        this.init([], [this._el_0], [], []);
        return null;
    };
    _View_Img0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = this.context._h;
        if (import2.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementStyle(this._el_0, 'height', ((this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_0) == null) ? null : this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_0).toString()));
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this.context._w;
        if (import2.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementStyle(this._el_0, 'width', ((this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_1) == null) ? null : this.viewUtils.sanitizer.sanitize(import11.SecurityContext.STYLE, currVal_1).toString()));
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Img0;
}(import3.AppView));
export function viewFactory_Img0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Img === null)) {
        (renderType_Img = viewUtils.createRenderComponentType('', 0, import12.ViewEncapsulation.None, styles_Img, {}));
    }
    return new _View_Img0(viewUtils, parentInjector, declarationEl);
}
